src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,204 @@
+                jsonWriter.WriteEndObject();
+            }
+        }
+
+        /// <summary>
+        /// Read the references from the project.lock.json file.
+        /// </summary>
+        internal static ImmutableArray<string> ReadProjectLockJson(string packagesDirectory, TextReader reader)
+        {
+            JObject obj;
+            using (var jsonReader = new JsonTextReader(reader))
+            {
+                obj = JObject.Load(jsonReader);
+            }
+            var builder = ArrayBuilder<string>.GetInstance();
+            var targets = (JObject)GetPropertyValue(obj, "targets");
+            foreach (var target in targets)
+            {
+                if (target.Key == ProjectLockJsonFramework)
+                {
+                    foreach (var package in (JObject)target.Value)
+                    {
+                        var packageRoot = PathUtilities.CombineAbsoluteAndRelativePaths(packagesDirectory, package.Key);
+                        var runtime = (JObject)GetPropertyValue((JObject)package.Value, "runtime");
+                        if (runtime == null)
+                        {
+                            continue;
+                        }
+                        foreach (var item in runtime)
+                        {
+                            var path = PathUtilities.CombinePossiblyRelativeAndRelativePaths(packageRoot, item.Key);
+                            builder.Add(path);
+                        }
+                    }
+                    break;
+                }
+            }
+            return builder.ToImmutableAndFree();
+        }
+
+        private static JToken GetPropertyValue(JObject obj, string propertyName)
+        {
+            JToken value;
+            obj.TryGetValue(propertyName, out value);
+            return value;
+        }
+
+        private void NuGetRestore(string projectJsonPath)
+        {
+            // Load nuget.exe from same directory as current assembly.
+            var nugetExePath = PathUtilities.CombineAbsoluteAndRelativePaths(
+                PathUtilities.GetDirectoryName(
+                    CorLightup.Desktop.GetAssemblyLocation(typeof(NuGetPackageResolverImpl).GetTypeInfo().Assembly)),
+                "nuget.exe");
+            var startInfo = new ProcessStartInfo()
+            {
+                FileName = nugetExePath,
+                Arguments = $"restore \"{projectJsonPath}\" -PackagesDirectory \"{_packagesDirectory}\"",
+                CreateNoWindow = true,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+            };
+            _restore(startInfo);
+        }
+
+        private static void NuGetRestore(ProcessStartInfo startInfo)
+        {
+            var process = Process.Start(startInfo);
+            string line;
+            var reader = process.StandardOutput;
+            while ((line = reader.ReadLine()) != null)
```

Comment:
This still risks deadlock -- if the process starts outputting standard error before standard output, you can still get hit. You need to use asynchrony here.